webview-ui/src/diagnostics_panel/controls/MinecraftStatisticTable.tsx

```diff
@@ -0,0 +1,166 @@
+}
+
+export enum MinecraftStatisticTableSortType {
+    Alphabetical,
+    Numerical,
+}
+
+type MinecraftStatisticTableProps = {
+    title: string;
+    statisticDataProvider: StatisticProvider;
+    statisticResolver: StatisticResolver;
+    defaultSortOrder?: MinecraftStatisticTableSortOrder;
+    defaultSortType?: MinecraftStatisticTableSortType;
+
+    keyLabel: string;
+    valueLabel: string;
+};
+
+const sortOrderOptions = [
+    { id: MinecraftStatisticTableSortOrder.Ascending, label: 'Ascending' },
+    { id: MinecraftStatisticTableSortOrder.Descending, label: 'Descending' },
+];
+
+const sortTypeOptions = [
+    { id: MinecraftStatisticTableSortType.Alphabetical, label: 'Alphabetical' },
+    { id: MinecraftStatisticTableSortType.Numerical, label: 'Numerical' },
+];
+
+export default function MinecraftStatisticTable({
+    title,
+    statisticDataProvider,
+    statisticResolver,
+    defaultSortOrder = MinecraftStatisticTableSortOrder.Descending,
+    defaultSortType = MinecraftStatisticTableSortType.Numerical,
+    keyLabel,
+    valueLabel,
+}: MinecraftStatisticTableProps): JSX.Element {
+    // states
+    const [data, setData] = useState<TrackedStat[]>([]);
+
+    const [selectedSortOrder, setSelectedSortOrder] = useState<MinecraftStatisticTableSortOrder>(defaultSortOrder);
+    const [selectedSortType, setSelectedSortType] = useState<MinecraftStatisticTableSortType>(defaultSortType);
+
+    const _onSelectedSortOrderChange = useCallback((e: Event | React.FormEvent<HTMLElement>): void => {
+        const target = e.target as HTMLSelectElement;
+        setSelectedSortOrder(sortOrderOptions[target.selectedIndex].id);
+    }, []);
+
+    const _onSelectedSortTypeChange = useCallback((e: Event | React.FormEvent<HTMLElement>): void => {
+        const target = e.target as HTMLSelectElement;
+        setSelectedSortType(sortTypeOptions[target.selectedIndex].id);
+    }, []);
+
+    useEffect(() => {
+        const eventHandler = (event: StatisticUpdatedMessage): void => {
+            // Update data with new data point
+            setData((prevState: TrackedStat[]): TrackedStat[] => {
+                let newData = statisticResolver(event, prevState);
+
+                // Calculate the latest tick for each category, as some content might come from different ticks
+                const latestTicks = new Map<string, number>();
+                newData.forEach(dataPoint => {
+                    const currentTick = latestTicks.get(dataPoint.category!) ?? 0;
+                    if (dataPoint.time > currentTick) {
+                        latestTicks.set(dataPoint.category!, dataPoint.time);
+                    }
+                });
+
+                // Filter out data points that are older than the latest tick
+                newData = newData.filter(dataPoint => {
+                    const latestTick = latestTicks.get(dataPoint.category!);
+                    return latestTick !== undefined && dataPoint.time === latestTick;
```

Comment:
Nit non-blocking, shouldn't be enough this?
```suggestion
                    return dataPoint.time === latestTick;
```